### PR TITLE
spisim_com_enhancement

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

File: pyaedt/generic/com_parameters.py [hunk 1]
```diff
@@ -45,6 +48,12 @@ def parameters(self):
     def load(self, file_path):
         """Load configuration file.
 
+
+        Parameters
+        ----------
+        file_path: str
+            Path of the configure file.
```
Comment (PipKat) [inline pyaedt/generic/com_parameters.py:55]:
```suggestion
            Path of the configuration file.
```

File: pyaedt/generic/com_parameters.py [hunk 2]
```diff
@@ -45,6 +48,12 @@ def parameters(self):
...
+
         Returns
         -------
         bool
```
Comment (PipKat) [inline pyaedt/generic/com_parameters.py:59]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/generic/com_parameters.py [hunk 3]
```diff
@@ -59,6 +68,65 @@ def load(self, file_path):
                     self.__setattr__(name, str(value))
         return True
 
+    @pyaedt_function_handler
+    def load_ieee_excel(self, file_path):
+        """Load IEEE original configure file in Excel format.
```
Comment (PipKat) [inline pyaedt/generic/com_parameters.py:73]:
```suggestion
        """Load the IEEE original configuration file in Excel format.
```

File: pyaedt/generic/com_parameters.py [hunk 4]
```diff
@@ -59,6 +68,65 @@ def load(self, file_path):
...
+
+        Parameters
+        ----------
+        file_path: str
+         Path of the Execel file.
+
+        Returns
+        -------
+        bool
```
Comment (PipKat) [inline pyaedt/generic/com_parameters.py:82]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/misc/spisim_com_configuration_files/00_COMSettingsExplained.csv [hunk 5]
```diff
@@ -0,0 +1,153 @@
+Notes:,,,"If Mablab_keyword is empty, it means same as SPISim_Keyword or replace '_' which space (Matlab is case sensitive)"
```
Comment (PipKat) [inline pyaedt/misc/spisim_com_configuration_files/00_COMSettingsExplained.csv:1]: Is Mablab a typo?

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+        elif keyword in self.receiver_testing:
+            self.receiver_testing[keyword] = value
+        elif keyword in self.spisim_control:
+            self.spisim_control[keyword] = value
+        else:
+            self.other_parameters[keyword] = value
+
+    @pyaedt_function_handler
+    def export(self, file_path):
+        """Export COM parameter to a json file.
```
Comment (SMoraisAnsys) [inline pyaedt/misc/spisim_com_configuration_files/com_parameters.py:124]:
```suggestion
        """Export COM parameter to a JSON file.
```

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py [hunk 7]
```diff
@@ -0,0 +1,375 @@
...
+        temp["receiver_testing"] = self.receiver_testing
+        temp["spisim_control"] = self.spisim_control
+        temp["other_parameters"] = self.other_parameters
+
+        with open(file_path, "w", encoding="utf-8") as f:
+            f.write(json.dumps(temp, indent=4, ensure_ascii=False))
+
+    @pyaedt_function_handler
+    def load(self, file_path):
+        """Load COM parameters from a json file.
```
Comment (SMoraisAnsys) [inline pyaedt/misc/spisim_com_configuration_files/com_parameters.py:151]:
```suggestion
        """Load COM parameters from a JSON file.
```

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+        with open(file_path, "r") as fp:
+            lines = fp.readlines()
+            for line in lines:
+                if not line.startswith("#") and "=" in line:
+                    split_line = [i.strip() for i in line.split("=")]
+                    kw, value = split_line
+                    if kw in reverse_map:  # Get Matlab keyword
+                        kw = reverse_map[kw]
+                    self.set_parameter(kw, value)
+        return
```
Comment (SMoraisAnsys) [inline pyaedt/misc/spisim_com_configuration_files/com_parameters.py:211]:
```suggestion
        return True
```
You should probably add a try except to return `False` if something goes wrong ?

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+        self.io_control.update(
+            {
+                # "DIAGNOSTICS": "",
+                # "DISPLAY_WINDOW": "",
+                # "CSV_REPORT": "",
+                "RESULT_DIR": "",
+                # "SAVE_FIGURES": "",
+                "Port Order": "",
+                "RUNTAG": "",
+                # "COM_CONTRIBUTION": "",
```
Comment (SMoraisAnsys) [inline pyaedt/misc/spisim_com_configuration_files/com_parameters.py:277]: Keep this comments ?

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py [hunk 10]
```diff
@@ -0,0 +1,375 @@
...
+            {
+                "COM Pass threshold": "",
+                "ERL Pass threshold": "",
+                "DER_0": "",
+                "T_r": "",
+                "FORCE_TR": "",
+                "Local Search": "",
+                # "BREAD_CRUMBS": "",
+                # "SAVE_CONFIG2MAT": "",
+                # "PLOT_CM": "",
```
Comment (SMoraisAnsys) [inline pyaedt/misc/spisim_com_configuration_files/com_parameters.py:290]: Keep this comments ?

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Really good job here. I'm glad to see that we move to something out of IronPython and can leverage pathlib and other packages.

If this feature is urgently requested I'll approve it after the changes. If it is not urgent, I'd advise to refactor it a bit by creating classes to hold parameters information (if possible). I don't know all the COM parameters but I assume that, depending on some information, they should contain come pre defined information. Moreover, I'd avoid using the following snippet which is quite confusing to read:
```python
        self._standard = standard
        self.standard = standard
```
Of course, that can be performed later on if needed

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py (hunk 9 above)
Comment (ring630) [reply pyaedt/misc/spisim_com_configuration_files/com_parameters.py:277]: keep them here for now. I am not sure if they will be used in the later enhancement.

File: pyaedt/misc/spisim_com_configuration_files/com_parameters.py (hunk 10 above)
Comment (ring630) [reply pyaedt/misc/spisim_com_configuration_files/com_parameters.py:290]: The same as above

**Comment by ring630** [comment]
> Really good job here. I'm glad to see that we move to something out of IronPython and can leverage pathlib and other packages.
> 
> If this feature is urgently requested I'll approve it after the changes. If it is not urgent, I'd advise to refactor it a bit by creating classes to hold parameters information (if possible). I don't know all the COM parameters but I assume that, depending on some information, they should contain come pre defined information. Moreover, I'd avoid using the following snippet which is quite confusing to read:
> 
> ```python
>         self._standard = standard
>         self.standard = standard
> ```
> 
> Of course, that can be performed later on if needed

I will refactor the parameter class later when I understand the standard better. Please approve.

**Review by SMoraisAnsys — APPROVED** [review]

**Pull request merged**